core/src/test/java/org/ehcache/EhcacheBasicPutAllEventsTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache;
+
+import static org.ehcache.EhcacheBasicBulkUtil.KEY_SET_A;
+import static org.ehcache.EhcacheBasicBulkUtil.KEY_SET_B;
+import static org.ehcache.EhcacheBasicBulkUtil.KEY_SET_C;
+import static org.ehcache.EhcacheBasicBulkUtil.KEY_SET_D;
+import static org.ehcache.EhcacheBasicBulkUtil.fanIn;
+import static org.ehcache.EhcacheBasicBulkUtil.getAltEntryMap;
+import static org.ehcache.EhcacheBasicBulkUtil.getEntryMap;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.events.CacheEventNotificationService;
+import org.ehcache.events.CacheEventNotificationServiceImpl;
+import org.ehcache.exceptions.CacheAccessException;
+import org.ehcache.function.Function;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.util.IsCreatedOrUpdated;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides events based testing for PUT_ALL operations
+ */
+public class EhcacheBasicPutAllEventsTest extends EhcacheEventsTestBase {
+
+  protected IsCreatedOrUpdated<String, String> isCreatedOrUpdated = new IsCreatedOrUpdated<String, String>();
+
+  @Test
+  public void testPutAllMapNull() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+    final Ehcache<String, String> ehcache = getEhcache();
```

Comment:
Again being picky, it seems that 'store' shouldn't be a class field since it isn't shared amongst multiple methods, so it should a local variable passed as an argument to getEhcache(...)
Otherwise, it is not clear why we need to init a Store